### ISSUE TITLE
pgi/19.4 task_fib_single and KokkosCore_PerformanceTest_TaskDag seg faults

### ISSUE THREAD

**Issue by ndellingwood** [kokkos/kokkos#2806]
I ran tests on kokkos-dev-2 with pgi/19.4 to check status before adding a new nightly test and I am seeing seg faults in the `task_fib_single` and `KokkosCore_PerformanceTest_TaskDag` tests with the OpenMP backend

Sample reproducer:

Kokkos SHA: 75eaf2e411a411faca806a7d479c07aace404687

```
module load sems-env kokkos-env sems-cmake/3.12.2 sems-gcc/7.3.0 pgi/19.4

<KOKKOS_PATH>/generate_makefile.bash --with-devices=OpenMP --arch=SNB,Volta70 --compiler=pgc++ --cxxflags="-O3 " --cxxstandard="11" --ldflags="" --kokkos-path=<KOKKOS_PATH> --no-examples --with-options=enable_large_mem_tests
```

Edit: Added SHA

**Comment by ndellingwood** [comment]
Adding @dhollman

**Comment by PhilMiller** [comment]
This is likely related to / a subset of #2582. Potentially to be addressed by #5748

**Comment by masterleinad** [comment]
We don't support PGI anymore.
